ci/gpu/notebook_list.py

```diff
@@ -25,6 +44,25 @@
 pascal = False
 ampere = False
 device = cuda.get_current_device()
+
+opts, args = getopt.getopt( sys.argv[1:],"r:", ["runtype"])
+
+runtype='ci'
+for opt, arg in opts:
+    if opt in ['-r',"--runtype"]:
+       runtype = arg
+    else:
+       print(f'Unknown argument  = {opt} = {arg}', file=sys.stderr)
+       exit()
+
+#if runtype not in ['ci', 'nightly', 'weekly']:
+#    print(f'Unknown Run Type  = {runtype}', file=sys.stderr)
+#    exit()
```

Comment:
Is this dead code that can be removed?

Reply:
This is meant to be a test for an unrecognized run type, I added the new default 'all' to the valid types and uncommented.
Changed run types and skip files to a dictionary and modified this check to check the dictionary